packages/video_player_avplay/tizen/src/drm_manager_proxy.cc

```diff
@@ -6,59 +6,103 @@
 
 #include <dlfcn.h>
 
-FuncDMGRSetData DMGRSetData = nullptr;
-FuncDMGRGetData DMGRGetData = nullptr;
-FuncDMGRSetDRMLocalMode DMGRSetDRMLocalMode = nullptr;
-FuncDMGRCreateDRMSession DMGRCreateDRMSession = nullptr;
-FuncDMGRSecurityInitCompleteCB DMGRSecurityInitCompleteCB = nullptr;
-FuncDMGRReleaseDRMSession DMGRReleaseDRMSession = nullptr;
-
-void* OpenDrmManagerProxy() { return dlopen("libdrmmanager.so.0", RTLD_LAZY); }
-
-int InitDrmManagerProxy(void* handle) {
-  if (!handle) {
-    return DM_ERROR_INVALID_PARAM;
+#include "log.h"
+
+DrmManagerProxy::DrmManagerProxy()
+    : drm_manager_handle_(::dlopen("libdrmmanager.so.0", RTLD_LAZY)) {
+  if (drm_manager_handle_ == nullptr) {
+    LOG_ERROR("Fail to open drm manager.");
+    return;
   }
 
-  DMGRSetData = reinterpret_cast<FuncDMGRSetData>(dlsym(handle, "DMGRSetData"));
-  if (!DMGRSetData) {
-    return DM_ERROR_DL;
+  dmgr_set_data_ = reinterpret_cast<FuncDMGRSetData>(
+      dlsym(drm_manager_handle_, "DMGRSetData"));
+  if (dmgr_set_data_ == nullptr) {
+    LOG_ERROR("Fail to find DMGRSetData.");
   }
 
-  DMGRGetData = reinterpret_cast<FuncDMGRGetData>(dlsym(handle, "DMGRGetData"));
-  if (!DMGRGetData) {
-    return DM_ERROR_DL;
+  dmgr_get_data_ = reinterpret_cast<FuncDMGRGetData>(
+      dlsym(drm_manager_handle_, "DMGRGetData"));
+  if (dmgr_get_data_ == nullptr) {
+    LOG_ERROR("Fail to find DMGRGetData.");
   }
 
-  DMGRSetDRMLocalMode = reinterpret_cast<FuncDMGRSetDRMLocalMode>(
-      dlsym(handle, "DMGRSetDRMLocalMode"));
-  if (!DMGRSetDRMLocalMode) {
-    return DM_ERROR_DL;
+  dmgr_set_drm_local_mode_ = reinterpret_cast<FuncDMGRSetDRMLocalMode>(
+      dlsym(drm_manager_handle_, "DMGRSetDRMLocalMode"));
+  if (dmgr_set_drm_local_mode_ == nullptr) {
+    LOG_ERROR("Fail to find DMGRSetDRMLocalMode.");
```

Comment:
Since all of these symbols are necessary, I think it is better to return immediately if any of them fail to load.

Reply:
Yes, good suggestion. I add a new method to obtaion address of drm manager functions.